snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/SnomedEditingContext.java

```diff
@@ -557,58 +557,67 @@ public void delete(EObject object, boolean force) {
 	}
 	
 	private void delete(Concept concept, boolean force) {
```

Comment:
With a single `SnomedDeletionPlan` you could remove the component-specific delete methods, keep the `canDelete(...)` methods and move the code that throws the rejection exception at the end of the public `delete(EObject, boolean)` method.